examples/ProjPlaneCurve.jl

```diff
@@ -0,0 +1,534 @@
+        push!(ro, -f + gens(rr)[1])
+     end
+  end
+  for y in ro
+     push!(Pts, Oscar.Geometry.ProjSpcElem(PP, [lc(y), R.R.base_ring(1), R.R.base_ring(0)]))
+  end
+  if evaluate(F, [1, 0, 0]) == 0 && evaluate(H, [1, 0, 0]) == 0
+     push!(Pts, Oscar.Geometry.ProjSpcElem(PP, [R.R.base_ring(1), R.R.base_ring(0), R.R.base_ring(0)]))
+  end
+  return [CC,Pts]
+end
+
+################################################################################
+# without specifying the projective space.
+
+@doc Markdown.doc"""
+curve_intersect(C::ProjPlaneCurve{S}, D::ProjPlaneCurve{S}) where S <: FieldElem
+> Returns a list whose first element is the projective plane curve defined by the gcd of F and G, the second element is the list of the remaining intersection points when the common components are removed from C and D (the points are in a new projective space).
+"""
+function curve_intersect(C::ProjPlaneCurve{S}, D::ProjPlaneCurve{S}) where S <: FieldElem
+   R = parent(C.eq)
+   PP = projective_space(R.R.base_ring, 2)
+   curve_intersect(PP[1], C, D)
+end
+
+################################################################################
+# Compute the singular locus of an affine plane curve by intersecting the
+# derivatives. The points might also be contained in the components.
+
+@doc Markdown.doc"""
+curve_singular_locus(PP::Oscar.Geometry.ProjSpc{S}, C::ProjPlaneCurve{S}) where S <: FieldElem
+> Returns a list whose first element is the projective plane curve consisting of the singular components of C (if any), and the second element is the list of the singular points of the reduction of C (the points are in PP). The singular component might not contain any point over the considered field.
+"""
+function curve_singular_locus(PP::Oscar.Geometry.ProjSpc{S}, C::ProjPlaneCurve{S}) where S <: FieldElem
+  D = reduction(C)
+  Pts = Array{Oscar.Geometry.ProjSpcElem, 1}()
+  CC = Array{ProjPlaneCurve, 1}()
+  # The components with multiplicity > 1 are singular
+  f = []
+  for h in keys(C.components)
+     if C.components[h] != 1
+        push!(f, h.eq^C.components[h])
+     end
+  end
+  if isempty(f) == false
+     g = prod(f)
+     push!(CC, ProjPlaneCurve(g))
+  end
+  # Computation of the singular locus of the reduction of C (which consists of only isolated points)
+  J = jacobi_ideal(D)
+  FX = gens(J)[1]
+  FY = gens(J)[2]
+  FZ = gens(J)[3]
+  if isconstant(FX) == true
+     if FX != 0
+        return [CC, Pts]
+     else
+        #FX = 0
+        if isconstant(FY) == true
+           if FY != 0
+             return [CC, Pts]
+           else
+              #FX = FY = 0
+              if isconstant(FZ) == true
+                 if FZ != 0
+                    return [CC, Pts]
+                 else
+                    # FX = FY = FZ = 0
+                    push!(CC, C)
+                    return [CC, Pts]
+                 end
+              else
+                 # FX = FY = 0 and FZ non constant
+                 return(CC, Pts)
```

Comment:
```suggestion
                 return (CC, Pts)
```
Also, maybe return in other places a tuple like here? It's nice to have a consistent return type.